guide/src/migration.md

```diff
@@ -47,6 +47,37 @@ However, take care to note that the behaviour is different from previous version
 Related to this, we also added a `pyo3_disable_reference_pool` conditional compilation flag which removes the infrastructure necessary to apply delayed reference count decrements implied by `impl<T> Drop for Py<T>`. They do not appear to be a soundness hazard as they should lead to memory leaks in the worst case. However, the global synchronization adds significant overhead to cross the Python-Rust boundary. Enabling this feature will remove these costs and make the `Drop` implementation abort the process if called without the GIL being held instead.
 </details>
 
+### Deprecation of implicit integer comparison for simple enums
+<details open>
+<summary><small>Click to expand</small></summary>
+
+With `pyo3` 0.22 the implicit implementation of integer comparison for simple enums using their discriminants is deprecated. To migrate, place a `#[pyo3(eq_int)]` attribute on affected classes. In addition the `#[pyo3(eq)]` option can be used to implement comparison based on the `PartialEq` implementation. If both options are specified, comparing by `PartialEq` and on failure the integer comparision is used as a fallback.
```

Comment:
It might be nice to phrase this in a way which makes it clear that we're not intending to remove the functionality, just make it explicit. Maybe something like the below?

```suggestion
### Require explicit opt-in for comparison for simple enums
<details open>
<summary><small>Click to expand</small></summary>

With `pyo3` 0.22 the new `#[pyo3(eq)]` options allows automatic implementation of Python equality using Rust `PartialEq`. Previously simple enums automatically implemented equality in terms of their discriminants. To make PyO3 more consistent this automatic equality implementation is deprecated in favour of having opt-ins for all `#[pyclass]` types. Similarly, simple enums supported comparison with integers, which is not covered by Rust's `PartialEq` derive so has been split out into an `eq_int` attribute.

To migrate, place a `#[pyo3(eq, eq_int)]` attribute on simple enum classes.
```

Reply:
Sounds good to me